src/legacy/server/sample_data/routes/list.js

```diff
@@ -24,13 +24,56 @@ const NOT_INSTALLED = 'not_installed';
 const INSTALLED = 'installed';
 const UNKNOWN = 'unknown';
 
+/*
+ * @return {Function} : gets the status for a sample data set
+ */
+const getStatusFactoryFn = request => async sampleDataset => {
```

Comment:
I always think having a function that returns a function just to scope parameters hurts readability in the long run. `getStatusFactoryFn` is kind of a meaningless name that hides the true action of the logic - which is to set the status for the sample data set.

Why not just move this function inside the handler so request is in scope?

Then `getStatusFactoryFn` can just be `setSampleDataSetStatus` and everything is much easier to read.

Or if you want to remove all of the logic from `handler`, why not abstract everything and create a method called `async setStatusForSampleDatasets (request, sampleDatasets)`? Then this function can include the `sampleDatasets.map` and fully contain the required scope and better encapsulate a single idea.

Reply:
I'm ++ for moving it back to the original code: the inline closure isn't a problem